implementations/python/sugar-lift-python-source/src/sugar_lift_python_source/value_pins.py

```
@@ -189,8 +192,12 @@ def scan_module_value_pins(
             scan.boundaries.append(_pin_boundary(candidate, exc.reason))
             mutable_kind = _direct_mutable_kind(candidate.value)
             if mutable_kind is not None:
+                if candidate.binding_occurrence is None:
+                    raise AssertionError("mutable global candidate lost binding occurrence")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)